crates/csslsrs/src/features/hover.rs

```diff
@@ -1,131 +1,110 @@
-use crate::css_data::{
-    AtDirectiveEntry, CssCustomData, MarkupDescriptionOrString, PropertyEntry, Reference, Status,
-};
-use biome_css_syntax::{CssLanguage, CssSyntaxKind};
-use biome_rowan::{AstNode, SyntaxNode};
-use lsp_types::{Hover, HoverContents, MarkupContent, MarkupKind, Position, TextDocumentItem};
-use std::fmt::Write;
-
 use crate::{
-    converters::{from_proto::offset, line_index::LineIndex, to_proto::range, PositionEncoding},
+    converters::{
+        from_proto::offset,
+        line_index::LineIndex,
+        to_proto::{self, range},
+        PositionEncoding,
+    },
+    css_data::{
+        AtDirectiveEntry, CssCustomData, MarkupDescriptionOrString, PropertyEntry, Reference,
+        Status,
+    },
     service::LanguageService,
 };
+use biome_css_syntax::{CssLanguage, CssSyntaxKind};
+use biome_rowan::{AstNode, SyntaxNode, TextSize};
+use lsp_types::{
+    Hover, HoverContents, MarkupContent, MarkupKind, Position, Range, TextDocumentItem,
+};
+use std::fmt::Write;
 
 /// Extracts hover information for the given CSS node and position.
 fn extract_hover_information(
     node: &SyntaxNode<CssLanguage>,
     position: Position,
     line_index: &LineIndex,
     encoding: PositionEncoding,
-    css_data: &Vec<&CssCustomData>,
+    css_data: &[&CssCustomData],
```

Comment:
?

Reply:
writing `&Vec` instead of `&[_]` involves a new object where a slice will do
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
`#[warn(clippy::ptr_arg)]` on by default